engine/src/test/java/org/camunda/bpm/engine/test/api/mgmt/ManagementServiceGetTelemetryDataTest.java

```diff
@@ -344,4 +348,81 @@ public void shouldResetCollectedMetricsDataWhenTelemetryEnabled() {
     assertThat(metrics.get(DECISION_INSTANCES).getCount()).isEqualTo(0);
     assertThat(metrics.get(EXECUTED_DECISION_ELEMENTS).getCount()).isEqualTo(0);
   }
+
+  @Test
+  public void shouldSetDataCollectionTimeFrameToEngineStartTimeWhenTelemetryDisabled() {
+    // given default telemetry data and empty telemetry registry
+    // current time after engine startup but before fetching telemetry data
+    Date beforeGetTelemetry = ClockUtil.getCurrentTime();
+    // move clock by one second to pass some time before fetching telemetry
+    ClockUtil.offset(1000L);
+
+    // when
+    TelemetryData telemetryData = managementService.getTelemetryData();
+
+    // then
+    assertThat(telemetryData.getProduct().getInternals().getDataCollectionStartDate()).isBefore(beforeGetTelemetry);
+  }
+
+  @Test
+  public void shouldNotResetCollectionTimeFrameAfterGetTelemetryWhenTelemetryDisabled() {
+    // given default telemetry data and empty telemetry registry
+    TelemetryData initialTelemetryData = managementService.getTelemetryData();
+
+    // when fetching telemetry data again
+    TelemetryData secondTelemetryData = managementService.getTelemetryData();
+
+    // then the data collection time frame should not reset after the first call
+    assertThat(initialTelemetryData.getProduct().getInternals().getDataCollectionStartDate()).isEqualTo(secondTelemetryData.getProduct().getInternals().getDataCollectionStartDate());
+  }
+
+  @Test
+  public void shouldSetDataCollectionTimeFrameToEngineStartTimeWhenTelemetryEnabled() {
+    // given default telemetry data and empty telemetry registry
+    // current time after engine startup but before fetching telemetry data
+    Date beforeGetTelemetry = ClockUtil.getCurrentTime();
+
+    // avtivate telemetry
```

Comment:
Small typo in a test comment
```suggestion
    // activate telemetry
```